src/System.Diagnostics.TraceSource/src/System/Diagnostics/DefaultTraceListener.cs

```diff
@@ -81,12 +107,28 @@ public override void Fail(string message)
         /// </devdoc>
         public override void Fail(string message, string detailMessage)
         {
-            // UIAssert is not enabled.
-            WriteAssert(string.Empty, message, detailMessage);
+            string stackTrace;
+            try
+            {
+                stackTrace = new StackTrace(fNeedFileInfo:true).ToString();
+            }
+            catch
+            {
+                stackTrace = "";
+            }
+            WriteAssert(stackTrace, message, detailMessage);
+            if (AssertUiEnabled)
+            {
+                s_provider.ShowDialog(stackTrace, message, detailMessage, "Assertion Failed");
```

Comment:
`s_provider.ShowDialog` is overriden to do nothing, right? What's the point of calling that method if it doesn't do anything?

Reply:
My idea was to provide the mechanism to customize behavior by adding `s_provider.ShowDialog`. It is currently empty because the code currently has nothing implemented for when `AssertUiEnabled` is true. Also note, `AssertUiEnabled` is true by default.

Perhaps I can add a comment on line 26 to explain this.

Reply:
Added the ShowDialog limitations in issue https://github.com/dotnet/corefx/issues/32955 and commented explanation in source code as well